src/campaign.rs

```diff
@@ -0,0 +1,109 @@
+use reqwest::header::{Authorization, Bearer, Header};
+use reqwest::Client;
+use serde_json::{self, Value};
+use std::io::{self, Write};
+use uuid::Uuid;
+
+use datatype::{Token, Url};
+use error::Error;
+
+const API_VERSION: &'static str = "api/v2";
+
+pub trait Campaign {
+    fn create(&self, campaign_id: Uuid, name: &str, groups: Vec<Uuid>) -> Result<(), Error>;
+    fn get(&self, campaign_id: Uuid) -> Result<(), Error>;
+    fn launch(&self, campaign_id: Uuid) -> Result<(), Error>;
+    fn stats(&self, campaign_id: Uuid) -> Result<(), Error>;
+    fn cancel(&self, campaign_id: Uuid) -> Result<(), Error>;
+}
+
+pub struct Manager<'c> {
+    client: &'c Client,
+    campaigner: Url,
+    token: Token,
+}
+
+impl<'c> Manager<'c> {
+    pub fn new(client: &'c Client, campaigner: Url, token: Token) -> Self {
+        Manager {
+            client,
+            campaigner,
+            token,
+        }
+    }
+
+    fn bearer(&self) -> Result<impl Header, Error> {
+        Ok(Authorization(Bearer {
+            token: (self.token)()?.access_token,
+        }))
+    }
+}
+
+impl<'c> Campaign for Manager<'c> {
+    fn create(&self, campaign_id: Uuid, name: &str, groups: Vec<Uuid>) -> Result<(), Error> {
```

Comment:
s/campaign_id/multi_target_update_id/

How do you create multi target updates btw?

Reply:
As discussed, I'll add the prior Director API call in another commit.